test/e2e/upgrades/storage/volume_mode.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+
+	"k8s.io/api/core/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const devicePath = "/mnt/volume1"
+
+// VolumeModeDowngradeTest tests that a VolumeMode Block PV is not mistakenly
+// formatted and mounted like a nil/Filesystem PV after a downgrade to a version
+// where the BlockVolume feature is disabled
+type VolumeModeDowngradeTest struct {
+	pvSource *v1.PersistentVolumeSource
+	pv       *v1.PersistentVolume
+	pvc      *v1.PersistentVolumeClaim
+	pod      *v1.Pod
+}
+
+func (VolumeModeDowngradeTest) Name() string {
+	return "[sig-storage] volume-mode-downgrade"
+}
+
+func (t *VolumeModeDowngradeTest) Skip(upgCtx upgrades.UpgradeContext) bool {
+	if !framework.ProviderIs("gce", "gke") {
+		return true
+	}
+
+	// Only run when downgrading from >= 1.13 to < 1.13
+	blockVersion := version.MustParseSemantic("1.13.0")
+	if upgCtx.Versions[0].Version.LessThan(blockVersion) {
+		return true
+	}
+	if !upgCtx.Versions[1].Version.LessThan(blockVersion) {
+		return true
+	}
+
+	return false
+}
+
+// Setup creates a block pv and then verifies that a pod can consume it.  The pod writes data to the volume.
+func (t *VolumeModeDowngradeTest) Setup(f *framework.Framework) {
+
+	var err error
+
+	cs := f.ClientSet
+	ns := f.Namespace.Name
+
+	By("Initializing PV source")
+	t.pvSource, _ = framework.CreateGCEVolume()
```

Comment:
Can we create a PVC using the default storageclass?  That would make the test portable across envs too.

Reply:
yes, done; i kept the provider check for the envs who have a default storageclass